File: day07/day07-input-output-01-ts/example2.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

// 建立 Zone
const zone = new aws.route53.Zone("example", {
    name: "example.com.",
});

const www = new aws.route53.Record("example", {
    zoneId: zone.zoneId,
    // 傳遞 Promise<string> 至 Input<string> 中
    name: Promise.resolve("www.example.com."),
    type: "TXT",
    records: [
        "Hello, World"
    ]
});

File: day07/day07-input-output-01-cs/Example2.cs
using System.Threading.Tasks;
using Pulumi;

namespace day07_input_output_01_cs;

using Aws = Pulumi.Aws;

public class Example2
{
    public static void Example2_()
    {
        // 建立 Zone
        var zone = new Aws.Route53.Zone("example", new Aws.Route53.ZoneArgs
        {
            Name = "example.com."
        });

        var www = new Aws.Route53.Record("example", new Aws.Route53.RecordArgs
        {
            ZoneId = zone.ZoneId,
            // 傳遞 async Task<string> 至 Input<string> 中
            // 因為 Output 是其中一種 Input 可以接受的型別，因此須先將 async task 轉成 Output
            Name = Output.Create(GetNameAsync()),
            Type = "TXT",
            Ttl = 300,
            Records = new[]
            {
                "Hello, World!"
            }
        });
    }

    private static async Task<string> GetNameAsync()
    {
        await Task.Delay(1);
        return "example.com.";
    }
}